src/main/java/Duke.java

```diff
@@ -74,9 +75,28 @@ public void run() {
                         tokens = parser.getDescription().split(" /at ");
                         ui.transmitMessage("Got it. I've added this task:");
                         this.tasks.add(new Events(tokens[0], tokens[1]));
-
+                        break;
+                    case "find":
+                        String keyWord = parser.getDescription();
+                        ArrayList<Task>  matchingTasks = new ArrayList<>();
+                        for(int i = 0 ; i < tasks.taskStorage.size() ; i++) {
+                            Task task = tasks.taskStorage.get(i);
+                            if((task.getDescription()).contains(keyWord)){
+                                matchingTasks.add(task);
+                            }
+                        }
+                        if(matchingTasks.size() == 0){
+                            ui.transmitMessage("There is no matching task in your list");
+                        }
+                        else {
+                            System.out.println("Here are the matching tasks in your list:");
+                            for (Task task : matchingTasks){
+                                System.out.println(task.toString());
```

Comment:
Since you override toString, is there a way to efficiently use it to print the messages instead of calling toString everytime?